packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -129,8 +134,8 @@ export function makeRumPublicApi(
     const startRumResults = startRumImpl(
       configuration,
       () => ({
-        user,
-        context: globalContextManager.get(),
+        user: userContextManager.getContext(),
+        context: globalContextManager.getContext(),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏on second look, if a customer relied on the fact that we did not clone the global context to pass some modifications by reference, it may be a breaking change, no? 
If so, it could be safer to keep using `globalContextManager.get()` until the next major.
wdyt?

Reply:
I believe in this case it is fine; this 2nd param in `startRumImpl` refers to `getCommonContext` in [`startRum`](https://github.com/DataDog/browser-sdk/blob/main/packages/rum-core/src/boot/startRum.ts#L28). 

That function is then passed to [`startRumAssembly`](https://github.com/DataDog/browser-sdk/blob/main/packages/rum-core/src/domain/assembly.ts#L63) which uses it to define the variable [`commonContext`](https://github.com/DataDog/browser-sdk/blob/main/packages/rum-core/src/domain/assembly.ts#L102). 

Finally that variable is used to set [`serverRumEvent.context`](https://github.com/DataDog/browser-sdk/blob/main/packages/rum-core/src/domain/assembly.ts#L138), which is in turn dispatched by [`lifeCycle.notify(LifeCycleEventType.RUM_EVENT_COLLECTED, serverRumEvent)`](https://github.com/DataDog/browser-sdk/blob/main/packages/rum-core/src/domain/assembly.ts#L152)

To my (hesitant) understanding I believe that is variable is not mutated or exposed publicly therefore making such a change ok. Thoughts?

Reply:
It could have been exposed through `beforeSend` in `event.context` but since there is a deep clone performed by `limitModification` it is not, so all good :swea

Reply:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Since it is internal usage only, and we know that it's not mutateted, maybe we can still use `globalContextManager.get()`? Using `getContext` doesn't seem useful here?

Reply:
@BenoitZugmeyer knowing we are looking  at deprecating `get` in contextManager, I think it makes sense to change this, even though this is only an internal API

Reply:
Nothing forces us to deprecate it (we might want to rename it to make it clearer though)

Reply:
I am not sure I follow? is this regarding the name `getContext`?

To recap:
- `contextManager.get()` is being phased out because it exposes the context. 
- `contextManager.getContext()` is now the preferred way of getting context, and returns a clone of the context (thus resolving the issue of the `get` method)